### PR TITLE
Cookierelease: Folder setup 1/3 (Excluding src and doc)

### PR DESCRIPTION
**Pull request by bobleesj**
Please review. @sbillinge  `src/` and `doc/` will be handled in the next PRs.

### CODE REVIEW

File: LICENSE.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+BSD 3-Clause License
+
+Copyright (c) 2009-2024, The Trustees of Columbia University
```
Comment (bobleesj) [inline LICENSE.rst:3]: 2009-2024 added

Comment (bobleesj) [inline LICENSE.rst:3]: 2009-2024 added

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+trying to commit again.
+
+Improvements and fixes are always appreciated.
+
+Before contribuing, please read our `Code of Conduct <https://github.com/diffpy/diffpy.nmf_mapping/blob/main/CODE_OF_CONDUCT.rst>`_.
+
+Contact
+-------
+
+For more information on diffpy.nmf_mapping please visit the project `web-page <https://diffpy.github.io/>`_ or email Prof. Simon Billinge at sb2896@columbia.edu.
```
Comment (bobleesj) [inline README.rst:121]: `diffpy.nmf_mapping` configured correctly.

**Review by bobleesj — COMMENTED** [review]
@sbillinge  Please review

File: LICENSE.rst (hunk 1 above)
Comment (sbillinge) [reply LICENSE.rst:3]: I think this work does not go back to 2009

**Comment by sbillinge** [comment]
Looks good. Just some issues with copyright dates maybe?

**Comment by bobleesj** [comment]
> Looks good. Just some issues with copyright dates maybe?

https://github.com/diffpy/diffpy.nmf_mapping/commit/d40b80700d907648b2622cfa240bda7bfb8d0182

I see the first commit being 2022. I will make a change.

File: LICENSE.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
+BSD 3-Clause License
+
+Copyright (c) 2022-2024, The Trustees of Columbia University
```
Comment (bobleesj) [inline LICENSE.rst:3]: I changed to 2022-2024 based on the first commit being Feb 2022

https://github.com/diffpy/diffpy.nmf_mapping/commit/d40b80700d907648b2622cfa240bda7bfb8d0182

**Review by bobleesj — COMMENTED** [review]
Please check the new copyright years

File: pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,53 @@
...
+]
+maintainers = [
+  { name="Simon J.L. Billinge group", email="simon.billinge@gmail.com" },
+]
+description = "Run NMF analysis on PDF and XRD data"
+keywords = ['diffpy PDF NMF']
+readme = "README.rst"
+requires-python = ">=3.10"
+classifiers = [
+        'Development Status :: 5 - Production/Stable',
```
[the remark this answers is not in the document]
Comment (sbillinge) [reply pyproject.toml:19]: https://gist.github.com/nazrulworld/3800c84e28dc464b2b30cec8bc1287fc

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+* LONGER DESCRIPTION HERE
+
+For more information about the diffpy.nmf_mapping library, please consult our `online documentation <https://diffpy.github.io/diffpy.nmf_mapping>`_.
+
+Citation
+--------
+
+If you use diffpy.nmf_mapping in a scientific publication, we would like you to cite this package as
+
+        diffpy.nmf_mapping Package, https://github.com/diffpy/diffpy.nmf_mapping
```
Comment (sbillinge) [inline README.rst:49]: I think there is a paper/papers that need citing for this one.  Also in the docs

**Review by sbillinge — COMMENTED** [review]
Please see comments

File: README.rst (hunk 5 above)
Comment (bobleesj) [reply README.rst]: Screenshot 2024-08-11 at 10 54 02 AM

Fixed in `REAMDE.rst`. I will add this in the following PR where I move from `docs` to `doc` using `git mv`. @sbillinge

Comment (bobleesj) [reply README.rst]: Citation source: https://pdfitc.org/NMF

**Comment by bobleesj** [comment]
@sbillinge  I also have a general GH workflow. 

Assume I made a mistake in a commit, and I want to revert my commit (to fix the problem). I also do not want the reviewer to waste time reading the mistake.

I see some options such as 1) force push 2) git revert. I see that `git revert` indicates a clear undoing of the previous changes. I feel uncomfortable with the word  "force".

What is the best practice?

File: README.rst (hunk 5 above)
Comment (bobleesj) [reply README.rst]: I see the citation format, in diffpy.structure and diffpy.pdffit2. I will make a change.

Screenshot 2024-08-11 at 11 04 40 AM

Screenshot 2024-08-11 at 11 04 48 AM

**Comment by bobleesj** [comment]
Please review:

Citations added to `README.rst`, URLs working. 

Screenshot 2024-08-11 at 11 17 50 AM

**Comment by sbillinge** [comment]
> Please review:
> 
> Citations added to `README.rst`, URLs working.
> 
> Screenshot 2024-08-11 at 11 17 50 AM

Nice job.  For the second paper add the words "and please consider citing " before it.

**Comment by bobleesj** [comment]
> Nice job. For the second paper add the words "and please consider citing " before it.

Yup. Added. I will reflect those in `doc` as well in the following PR.

**Comment by sbillinge** [comment]
> @sbillinge I also have a general question on a GH workflow.
> 
> Assume I made a mistake in a commit, and I want to revert my commit (to fix the problem). I also do not want the reviewer to waste time reading the mistake.
> 
> I see some options such as 1) force push 2) git revert. I see that `git revert` indicates a clear undoing of the previous changes. I feel uncomfortable with the word "force".
> 
> What is the best practice?

never force push.  I don't want to merge any force-pushes.  Some repo's have this set as a rule.   just revert if it is a small commit.  The revert applies the anti-change, so the change is done then undone, which is ok if it is small, but if it is big it is generally better to kill the branch and do a new PR.   Again, another reason to keep the edits on each PR to be small.

**Comment by bobleesj** [comment]
> never force push. I don't want to merge any force-pushes. Some repo's have this set as a rule. just revert if it is a small commit. The revert applies the anti-change, so the change is done then undone, which is ok if it is small, but if it is big it is generally better to kill the branch and do a new PR. Again, another reason to keep the edits on each PR to be small.

Thank you.  Well noted. @sbillinge

**Pull request merged**